src/main/java/org/openmainframeproject/cobolcheck/workers/Generator.java

```diff
@@ -210,4 +226,29 @@ private void closeReadersAndWriters(String programName) {
         writerController.closeWriter(programName);
     }
 
+    private void writeWhenOtherMockedSection(String sourceLine)  throws IOException{
```

Comment:
Hi Issac,
I think this method would fit better within the responsibility of the `TestSuiteParserController`. It could be renamed to `generateWhenOtherMockedSection`, and return a list of the lines that it has generated. The `Generator` class can then be responsible for calling `writeLines()` on the list.